pkg/authorization/controller/authorizationsync/normalize.go

```diff
@@ -3,16 +3,141 @@ package authorizationsync
+
+	return equivalentClusterRoleBinding, nil
+}
+
+func PrepareForUpdateClusterRoleBinding(equivalentClusterRoleBinding, rbacClusterRoleBinding *rbac.ClusterRoleBinding) bool {
+	// if we might need to update, we need to stomp fields that are never going to match like uid and creation time
+	normalizeObjectMeta(&equivalentClusterRoleBinding.ObjectMeta, rbacClusterRoleBinding.ObjectMeta)
+
+	// determine if they are equal
+	return apiequality.Semantic.DeepEqual(equivalentClusterRoleBinding, rbacClusterRoleBinding)
+}
+
+func PrepareForCreateRole(originRole *authorizationapi.Role) (*rbac.Role, error) {
+	// convert the origin role to an rbac role and compare the results
+	convertedRole := &rbac.Role{}
+	if err := authorizationapi.Convert_api_Role_To_rbac_Role(originRole, convertedRole, nil); err != nil {
+		return nil, err
+	}
+	// do a deep copy here since conversion does not guarantee a new object.
+	equivalentRole := &rbac.Role{}
+	if err := rbac.DeepCopy_rbac_Role(convertedRole, equivalentRole, cloner); err != nil {
+		return nil, err
+	}
+
+	// normalize rules before persisting so RBAC's case sensitive authorizer will work
+	normalizePolicyRules(equivalentRole.Rules)
+
+	// resource version cannot be set during creation
+	equivalentRole.ResourceVersion = ""
+
+	return equivalentRole, nil
+}
+
+func PrepareForUpdateRole(equivalentRole, rbacRole *rbac.Role) bool {
+	// if we might need to update, we need to stomp fields that are never going to match like uid and creation time
+	normalizeObjectMeta(&equivalentRole.ObjectMeta, rbacRole.ObjectMeta)
+
+	// determine if they are equal
+	return apiequality.Semantic.DeepEqual(equivalentRole, rbacRole)
+}
+
+func PrepareForCreateRoleBinding(originRoleBinding *authorizationapi.RoleBinding) (*rbac.RoleBinding, error) {
+	// convert the origin roleBinding to an rbac roleBinding and compare the results
+	convertedRoleBinding := &rbac.RoleBinding{}
+	if err := authorizationapi.Convert_api_RoleBinding_To_rbac_RoleBinding(originRoleBinding, convertedRoleBinding, nil); err != nil {
+		return nil, err
+	}
+	// do a deep copy here since conversion does not guarantee a new object.
+	equivalentRoleBinding := &rbac.RoleBinding{}
+	if err := rbac.DeepCopy_rbac_RoleBinding(convertedRoleBinding, equivalentRoleBinding, cloner); err != nil {
+		return nil, err
+	}
+
+	// resource version cannot be set during creation
+	equivalentRoleBinding.ResourceVersion = ""
+
+	return equivalentRoleBinding, nil
+}
+
+func PrepareForUpdateRoleBinding(equivalentRoleBinding, rbacRoleBinding *rbac.RoleBinding) bool {
+	// if we might need to update, we need to stomp fields that are never going to match like uid and creation time
+	normalizeObjectMeta(&equivalentRoleBinding.ObjectMeta, rbacRoleBinding.ObjectMeta)
+
+	// determine if they are equal
+	return apiequality.Semantic.DeepEqual(equivalentRoleBinding, rbacRoleBinding)
+}
+
+func normalizeObjectMeta(a *v1.ObjectMeta, b v1.ObjectMeta) {
```

Comment:
godoc explaining why you're doing this to these particular fields. Also, this could all be non mutating given shallow copies.  Take value objects and return a value object.